python/cugraph/cugraph/tests/test_sorensen.py

```diff
@@ -0,0 +1,239 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import gc
+import time
+import pytest
+
+import cudf
+import cugraph
+from cugraph.tests import utils
+from pathlib import PurePath
+
+# Temporarily suppress warnings till networkX fixes deprecation warnings
+# (Using or importing the ABCs from 'collections' instead of from
+# 'collections.abc' is deprecated, and in 3.8 it will stop working) for
+# python 3.7.  Also, this import networkx needs to be relocated in the
+# third-party group once this gets fixed.
+import warnings
+
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore", category=DeprecationWarning)
+    import networkx as nx
+
+
+print("Networkx version : {} ".format(nx.__version__))
+
+
+def cugraph_call(cu_M, edgevals=False):
+    G = cugraph.Graph()
+    if edgevals is True:
+        G.from_cudf_edgelist(cu_M, source="0", destination="1", edge_attr="2")
+    else:
+        G.from_cudf_edgelist(cu_M, source="0", destination="1")
+
+    # cugraph sorensen Call
+    t1 = time.time()
+    df = cugraph.sorensen(G)
+    t2 = time.time() - t1
+    print("Time : " + str(t2))
+
+    df = df.sort_values(["source", "destination"]).reset_index(drop=True)
+
+    return (
+        df["source"].to_array(),
+        df["destination"].to_array(),
+        df["sorensen_coeff"].to_array(),
+    )
+
+
+def networkx_call(M):
+
+    sources = M["0"]
+    destinations = M["1"]
+    edges = []
+    for i in range(len(M)):
+        edges.append((sources[i], destinations[i]))
+        edges.append((destinations[i], sources[i]))
+    edges = list(dict.fromkeys(edges))
+    edges = sorted(edges)
+    # in NVGRAPH tests we read as CSR and feed as CSC, so here we doing this
+    # explicitly
+    print("Format conversion ... ")
+
+    Gnx = nx.from_pandas_edgelist(
+        M, source="0", target="1", edge_attr="weight", create_using=nx.Graph()
+    )
+
+    # Networkx Jaccard Call
+    print("Solving... ")
+    t1 = time.time()
+    preds = nx.jaccard_coefficient(Gnx, edges)
+    t2 = time.time() - t1
+
+    print("Time : " + str(t2))
+    src = []
+    dst = []
+    coeff = []
+    for u, v, p in preds:
+        src.append(u)
+        dst.append(v)
+        # Conversion from Networkx Jaccard to Sorensen
+        # No networkX equivalent
+        coeff.append((2*p)/(1+p))
+    return src, dst, coeff
+
+
+@pytest.mark.parametrize("graph_file", utils.DATASETS_UNDIRECTED)
+def test_sorensen(graph_file):
+    gc.collect()
```

Comment:
I don't think our tests need to be calling this (I know it's been a pattern copied around a lot), but if we really don't want to remove this, we can at least make it a fixture, like done [here](https://github.com/rapidsai/cugraph/blob/df2d81dbb26904775386a6a25f20b776cd1897cb/python/cugraph/cugraph/tests/test_bfs.py#L76).

Reply:
Fixed